### PR TITLE
Add jitter to Ona GPS location

### PR DESCRIPTION
**Pull request by rudigiesler**

### CODE REVIEW

**Comment by rudigiesler** [comment]
@hodgestar The circle jitter algorithm is in. Have a look at it, and see if my thinking is correct.

File: go-app.js [hunk 1]
```diff
@@ -275,6 +276,23 @@ go.app = function() {
             });
         };
 
+        self.calculate_gps_offsets = function(toilet_code) {
+        // This function calculates the required GPS offsets given the
+        // toilet_code string
+            var cluster_len = self.im.config.cluster_len;
+            var issue_len = self.im.config.issue_len;
```
Comment (hodgestar) [inline go-app.js]: Could we add explicit defaults here for the lengths? Although I suspect `undefined * x == 0` in Javascript anyway. :)

Comment (hodgestar) [inline go-app.js]: Apparently the result is actually `NaN`. :)

Comment (rudigiesler) [inline go-app.js]: I'll add defaults of 0

File: test/app.test.js [hunk 2]
```diff
@@ -65,7 +65,9 @@ describe("App", function() {
                     username: 'root',
                     password: 'toor',
                     url: 'http://ona.io/api/v1/'
-                }
+                },
+                cluster_len: '0',
+                issue_len: '0'
```
Comment (hodgestar) [inline test/app.test.js]: These should probably be numerical zeroes rather than the character `0`.

File: test/app.test.js [hunk 3]
```diff
@@ -96,6 +98,42 @@ describe("App", function() {
         });
     });
 
+    describe("app.calculate_gps_offsets", function() {
+        describe("when cluster_len is set", function() {
+            it("should create deterministic offsets", function() {
+                app.im.config.cluster_len = '1';
+                app.im.config.issue_len = '0';
```
Comment (hodgestar) [inline test/app.test.js]: Here too.

**Comment by hodgestar** [comment]
Otherwise looks good!

File: test/app.test.js (hunk 2 above)
Comment (rudigiesler) [inline test/app.test.js]: Node doesn't seem to care

```
> '3' * 4
12
```

But I'll change it anyway, just in case.

**Comment by rudigiesler** [comment]
@hodgestar changes are done

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**